carball/analysis/kickoff_detection/kickoff_analysis.py

```diff
@@ -0,0 +1,159 @@
+from ...analysis.hit_detection.base_hit import BaseHit
+from ...analysis.simulator.ball_simulator import BallSimulator
+from ...analysis.simulator.map_constants import *
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseKickoff:
+
+    def is_kickoff(data_frame, index):
+        ball_position = data_frame.ball.loc[index, position_column_names]
+        if ((ball_position.pos_x == 0.0) and
+            (ball_position.pos_y == 0.0)): #z=92.74
+          return True
+        else:
+          return False
+
+    @staticmethod
+    def get_kickoffs_from_game(game: Game, proto_game: game_pb2, id_creation:Callable,
+                               player_map: Dict[str, Player],
+                               data_frame: pd.DataFrame, kickoff_frames: pd.DataFrame) -> Dict[int, KickoffStats]:
+        kickoffs = dict()
+        for frame in kickoff_frames:
+            cur_kickoff = proto_game.kickoff_stats.add()
+            end_frame = frame
+            while BaseKickoff.is_kickoff(data_frame, end_frame):
+                end_frame = end_frame + 1
+            cur_kickoff.touch
+            cur_kickoff.startframe = frame
+            cur_kickoff.touchframe = end_frame
+            for player in player_map.values():
+                kPlayer = cur_kickoff.touch.players.add()
+                kPlayer.player.id = player.id.id
+                kPlayer.kpos = BaseKickoff.get_kickoff_pos(player, data_frame, frame)
+                kPlayer.tpos = BaseKickoff.get_touch_pos(player, data_frame, frame, end_frame)
+                kPlayer.boost = data_frame[player.name]['boost'][end_frame]
+                BaseKickoff.set_jumps(kPlayer, player, data_frame, frame, end_frame)
+            cur_kickoff.type = BaseKickoff.get_kickoff_type(cur_kickoff.touch.players)
+            kickoffs[frame] = cur_kickoff
+        return kickoffs
+
+    @staticmethod
+    def set_jumps(kPlayer, player, data_frame, frame, end_frame):
+        ja  = data_frame[player.name]['jump_active']
+        dja = data_frame[player.name]['double_jump_active']
+        time = 0.0
+        firstBall = True
+        # check the kickoff frames (and then some) for jumps & big boost collection
+        for f in range(frame, end_frame + 20):
+            time = time + data_frame['game']['delta'][f]
+            if(firstBall and BaseKickoff.get_dist(data_frame, player.name, f) < 700):
+                kPlayer.ballTime = time
+                firstball = False
+            if (data_frame[player.name]['boost_collect'][f] == True):
+                kPlayer.boostTime = time
+            if (ja[f] != ja[f-1] or dja[f] != dja[f-1]):
+                kPlayer.jumps.append(time)
+
+    @staticmethod
+    def get_kickoff_type(players: list):
+        #
+        diagonals = [player.kpos for player in players].count(0)
+        offcenter = [player.kpos for player in players].count(1)
+        goalies   = [player.kpos for player in players].count(2)
+        if (len(players) == 6):
```

Comment:
in python you do not use parentheses in if statements unless it spans multiple lines.